_posts/2017-08-08-euroclojure-2017.md

```diff
@@ -0,0 +1,209 @@
+
+
+https://hackernoon.com/feynman-machine-a-new-approach-for-cortical-and-machine-intelligence-5855c0e61a70
+https://arxiv.org/pdf/1609.03971.pdf
+
+### Building a collaborative web app with ClojureScript, Gijs Stuurman
+
+Gijs has created a very nice collaborative (think google docs) Clojure evaluation environment called CREPL. It allows two or more people to edit the same Clojure code and evaluate it live in the browser. The interesting part was an integration to Reagent atoms that keeps the atom value in sync with the text in the buffer, so even the app state can be shared between participants.
+
+The later part of the talk was about the algorithm called Operational Transformation (OT) that is used to implement the collaborative editing part and how to test such an algorithm in a generative way.
+
+![Testing Operational Transformation](/img/euroclojure2017/testing-ot.png)
+
+### Clojure Tools for Symbolic Artificial Intelligence, Saul Johnson
+
+Symbolic artificial intelligence uses a human readable representation for problems and logic. Symbolic pattern matching, operator search and legal move generators are important concepts in this research area, and this talk presented a couple of Clojure libraries for these. With them, you can use STRIPS (Stanford Research Institute Problem Solver) style operators to specify rules and matcher functions to apply them to a set of data.
+
+
+![Pattern matching using matcher library](/img/euroclojure2017/symbolic-ai.png)
+
+### Dependency Heaven, Alex Miller
+
+Alex Miller’s talk was about the obstacles on getting Clojure 1.9 out the door and improving first time use of Clojure for beginners. He started with a recap on Rich Hickey’s keynote from last years Clojure/conj; one point that appeared in conversations to follow was that giving new names to avoid breakage is an underused solution. Also while maven itself is a great service, Clojure doesn’t really care about bytecode artifacts, since it ships with a compiler.
+
+With the separation of Clojure core into multiple artifacts, meaning you can’t run Clojure with a simple “java -cp clojure.jar clojure.main” command anymore, there’s an avenue to rework dependency resolution :). At the same time Clojure is getting an installer, a library for dependency resolution (tools.deps.alpha https://github.com/clojure/tools.deps.alpha) and a startup script (so you can ditch ~/bin/clj :)) which is to be published with brew, apt etc..
+
+
+Tools.deps.alpha resolves dependencies from a deps.edn file, which contains a map, not a vector of dependencies. This means that ordering of dependencies for resolving what version of a dependency to choose is now over and tools.deps.alpha takes care of this. The clj script then reads deps.edn and so could existing tools too. It remains to be seen how this will take on in tooling, but there definitely is a place for fresh take on dependency management here.
```

Comment:
Toka virke on aika pitkä ja vaikea seurata, ehkä voisi vähän tiivistää? Ehkä:  

This means that ordering of dependencies to indicate desired dependency version is over, now taken care by tools.deps.alpha.